### PR TITLE
Apply redux recompose

### PR DESCRIPTION
**Pull request by betoobernal**
## Summary

* adapt screen components to redux-recompose
* adapt reducers to redux-recompose
* adapt actions to redux-recompose

## Screenshots

[Change!] Show the screenshots of the views you modified.

## Trello Card

https://trello.com/c/u9l58025/10-implementar-redux-recompose-en-los-reducers

### CODE REVIEW

File: src/redux/Team/actions.js [hunk 1]
```diff
@@ -1,33 +1,69 @@
...
+  target: 'creationTeam',
+  service: TeamService.createTeam,
+  payload: values,
+  injections: [
+    withPostSuccess(dispatch => {
+      dispatch(teamCreated(true));
+    }),
+    withPostFailure(dispatch => {
+      dispatch(teamCreated(false));
+      throw new SubmissionError({ _error: 'Error...' });
```
Comment (betoobernal) [inline src/redux/Team/actions.js]: Venga, que esto lo puse para cargar el error de la petición al form. Pero aún no pongo el error que es porque no sé si va lo que devuelve el back o simplemente decir que no se pudo realizar la acción.
psd: Perdón por el español.

File: src/app/screens/AddMemberToTeam/index.js [hunk 2]
```diff
@@ -76,10 +76,10 @@ const mapDispatchToProps = dispatch => ({
         techs.push(tech);
       }
     });
-    /* eslint-disable no-param-reassign */
-    values = { ...values, techs };
-    return dispatch(teamActions.addMembersToTeam(values));
-  }
+    return dispatch(teamActions.addMembersToTeam({ ...values, techs }));
```
Comment (FrankIglesias) [inline src/app/screens/AddMemberToTeam/index.js]: check how to refactor this function later

File: src/app/screens/AddOwner/layout.js [hunk 3]
```diff
@@ -40,6 +40,7 @@ function AddOwner({ handleSubmit, ownerAdded, data, loading }) {
               </span>
             </div>
             <Field name="repository" component="select" className="form-control selectpicker">
+              <option />
```
Comment (FrankIglesias) [inline src/app/screens/AddOwner/layout.js]: add an empty text to the default option

File: src/redux/Repository/actions.js [hunk 4]
```diff
@@ -1,45 +1,92 @@
...
+  target: 'repoCreation',
+  service: RepositoryService.createRepository,
+  payload: values,
+  injections: [
+    withPostSuccess(dispatch => {
+      dispatch(repoCreated(true));
+    }),
+    withPostFailure(dispatch => {
+      dispatch(repoCreated(false));
+      throw new SubmissionError({ _error: 'Error...' });
```
Comment (FrankIglesias) [inline src/redux/Repository/actions.js]: use i18n later

File: src/redux/Repository/actions.js [hunk 5]
```diff
@@ -1,45 +1,92 @@
...
+  target: 'addMember',
+  service: RepositoryService.addMemberToOrg,
+  payload: values,
+  injections: [
+    withPostSuccess(dispatch => {
+      dispatch(memberAdded(true));
+    }),
+    withPostFailure(dispatch => {
+      dispatch(memberAdded(false));
+      throw new SubmissionError({ _error: 'Error...' });
```
Comment (FrankIglesias) [inline src/redux/Repository/actions.js]: same

File: src/redux/Repository/actions.js [hunk 6]
```diff
@@ -1,45 +1,92 @@
...
+  target: 'addCodeOwner',
+  service: RepositoryService.addOwnerToRepo,
+  payload: values,
+  injections: [
+    withPostSuccess(dispatch => {
+      dispatch(codeOwnerAdded(true));
+    }),
+    withPostFailure(dispatch => {
+      dispatch(codeOwnerAdded(false));
+      throw new SubmissionError({ _error: 'Error...' });
```
Comment (FrankIglesias) [inline src/redux/Repository/actions.js]: same

File: src/redux/Repository/actions.js [hunk 7]
```diff
@@ -1,45 +1,92 @@
...
+const addOwnerToRepository = values => ({
+  type: actions.ADD_CODE_OWNER,
+  target: 'addCodeOwner',
+  service: RepositoryService.addOwnerToRepo,
+  payload: values,
+  injections: [
+    withPostSuccess(dispatch => {
+      dispatch(codeOwnerAdded(true));
+    }),
+    withPostFailure(dispatch => {
```
Comment (FrankIglesias) [inline src/redux/Repository/actions.js]: add some kind of descriptive error

File: src/redux/Team/actions.js [hunk 8]
```diff
@@ -1,33 +1,69 @@
...
+  target: 'addMember',
+  service: TeamService.addMembersToTeam,
+  payload: values,
+  injections: [
+    withPostSuccess(dispatch => {
+      dispatch(memberAdded(true));
+    }),
+    withPostFailure(dispatch => {
+      dispatch(memberAdded(false));
+      throw new SubmissionError({ _error: 'Error...' });
```
Comment (FrankIglesias) [inline src/redux/Team/actions.js]: same

File: src/redux/Team/actions.js (hunk 1 above)
Comment (FrankIglesias) [reply src/redux/Team/actions.js]: - Return a more descriptive error like `Hubo un error en la creación del team`

File: src/services/RepositoryService.js [hunk 9]
```diff
@@ -1,35 +1,33 @@
...
-  throw response;
-};
-
-export const addOwnerToRepo = async values => {
-  const response = await api.post(`/repositories/${values.repository}/codeowners/`, {
+import { create } from 'apisauce';
+
+import api from '../config/api';
+
+/* export const createRepository = values => api.post('/repositories', values);
```
Comment (FrankIglesias) [inline src/services/RepositoryService.js]: delete commented lines if you are not using it

File: src/services/RepositoryService.js [hunk 10]
```diff
@@ -1,35 +1,33 @@
...
+
+export const getRepositories = () => api.get('/repositories?limit=100&page=2');
+
+export const createRepository = values =>
+  create({
+    baseURL: 'https://jsonplaceholder.typicode.com',
+    timeout: 55000
+  }).post('/posts', values);
+
+export const addMemberToOrg = values =>
```
Comment (FrankIglesias) [inline src/services/RepositoryService.js]: if you are going to mock de api, change the whole base url instead of changin for each function

Comment (FrankIglesias) [reply src/services/RepositoryService.js]: Move base URL to .env file

File: src/services/TeamServices.js [hunk 11]
```diff
@@ -1,27 +1,25 @@
...
-export const getTeams = async () => {
-  const response = await api.get('/teams/');
-  if (response.ok) {
-    return response.data;
-  }
-  throw response;
-};
+export const getTeams = () => api.get('/teams/');
+
+export const createTeam = values =>
```
Comment (FrankIglesias) [inline src/services/TeamServices.js]: same

File: src/services/TeamServices.js [hunk 12]
```diff
@@ -1,27 +1,25 @@
...
+}; */
 
-export const getTeams = async () => {
-  const response = await api.get('/teams/');
-  if (response.ok) {
-    return response.data;
-  }
-  throw response;
-};
+export const getTeams = () => api.get('/teams/');
```
Comment (FrankIglesias) [inline src/services/TeamServices.js]: extra slash at the end

**Review by FrankIglesias — CHANGES_REQUESTED** [review]

File: src/app/components/AlertInfo/index.js [hunk 13]
```diff
@@ -0,0 +1,27 @@
...
+}
+
+AlertInfo.propTypes = {
+  icon: PropTypes.string,
+  message: PropTypes.string
+};
+
+AlertInfo.defaultProps = {
+  icon: 'check',
+  message: '¡El registro se ha guardado con exito!'
```
Comment (FrankIglesias) [inline src/app/components/AlertInfo/index.js]: i18n

File: src/app/screens/AddOwner/layout.js [hunk 14]
```diff
@@ -33,12 +32,11 @@ function AddOwner({ handleSubmit, ownerAdded, data, loading }) {
             />
           </div>
           <div className="input-group">
-            <div className="input-group-prepend">
-              <span className="input-group-text">
-                <i className="material-icons">list</i>
-              </span>
-            </div>
+            <span className="input-group-text">
```
Comment (FrankIglesias) [inline src/app/screens/AddOwner/layout.js]: why this span? why not append classes to the i tag

File: src/app/screens/Teams/layout.js [hunk 15]
```diff
@@ -57,16 +56,7 @@ function TeamCreation({ handleSubmit, isError, loading, teamCreated }) {
...
-            <div className="alert alert-success">
-              <div className="container">
-                <div className="alert-icon">
-                  <i className="material-icons">check</i>
-                </div>
-                ¡El equipo se creó con éxito!
-              </div>
-            </div>
-          )}
+          {teamCreated && !loading && <AlertInfo message="¡El equipo se creó con éxito!" />}
```
Comment (FrankIglesias) [inline src/app/screens/Teams/layout.js]: i18n

**Review by FrankIglesias — APPROVED** [review]

**Pull request merged**